src/main/java/seedu/todo/ui/InputHandler.java

```diff
@@ -1,12 +1,95 @@
 package seedu.todo.ui;
 
+import java.util.LinkedList;
+
 import seedu.todo.controllers.*;
 
 public class InputHandler {
+    
+    private static InputHandler instance;
+    
+    private static final int MAX_HISTORY_SIZE = 20;
+    private static final int INITIAL_HISTORY_INDEX = -1;
+    private static LinkedList<String> commandHistory = new LinkedList<String>();
```

Comment:
ArrayDeque doesn't have `get` methods, and Deque doesnt specify `get` in the interface either. I'm trying with ListIterator now.